src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -1941,6 +1962,56 @@ private List<T> FilterModuleCollection<T>(IEnumerable<T> moduleCollection)
             return filteredModuleCollection;
         }
 
+        private void PrepareNoClobberWinCompatModuleImport(string moduleName, ModuleSpecification moduleSpec, ref ImportModuleOptions importModuleOptions)
+        {
+            List<string> NoClobberModuleList = PowerShellConfig.Instance.GetWindowsPowerShellCompatibilityNoClobberModuleList();
+
+            // moduleName can be just a module name and it also can be a full path to psd1 from which we need to extract the module name
+            var coreModuleToLoad = moduleName;
+            if (moduleSpec != null)
+            {
+                coreModuleToLoad = moduleSpec.Name;
+            }
+            else if (Path.IsPathRooted(moduleName))
+            {
+                coreModuleToLoad = Path.GetFileNameWithoutExtension(moduleName);
+            }
+
+            var coreModuleToLoadIsEngineModule = InitialSessionState.IsEngineModule(coreModuleToLoad);
+            if (coreModuleToLoadIsEngineModule || ((NoClobberModuleList != null) && NoClobberModuleList.Contains(coreModuleToLoad, StringComparer.OrdinalIgnoreCase)))
+            {
+                // if it is one of engine modules - first try to load it from $PSHOME\Modules
+                // otherwise rely on $env:PSModulePath (in which WinPS module location has to go after CorePS module location)
+                if (coreModuleToLoadIsEngineModule)
+                {
+                    var expectedCoreModulePath = Path.Combine(ModuleIntrinsics.GetPSHomeModulePath(), coreModuleToLoad);
```

Comment:
```suggestion
                    string expectedCoreModulePath = Path.Combine(ModuleIntrinsics.GetPSHomeModulePath(), coreModuleToLoad);
```

Reply:
(The type of the LHS is not explicitly inferable from the RHS, and also has a predefined short type name)

Reply:
updated.